app/controllers/cafe_controller.rb

```diff
@@ -0,0 +1,41 @@
+# encoding:UTF-8
+class CafeController < ApplicationController
+  skip_authorization
+
+  def index
+  end
+
+  def competition
+    lp = params[:lp] || '1'
+    year = Time.zone.now
+    if params[:year].present?
+      year = Time.zone.local(params[:year])
+    end
+
+    @competition = CafeCompetition.new(CafeQueries.cafe_workers(lp, year),
+                                       CafeQueries.working_users(lp, year),
+                                       lp, year)
+  end
+
+  def ladybug
+    authorize!(:ladybug, CafeShift)
+
+    if date = ladybug_date
```

Comment:
I do not understand this comment, because this is what I ended up with when trying to fix it...